amigocloud/amigocloud.py

```diff
@@ -73,22 +88,30 @@ def build_url(self, url):
         # User wants to use the api_url
         if url.startswith('/'):
             return self.api_url + url
-        return '%s/%s' % (self.api_url, url)
+        return os.path.join(self._project_url or self.api_url, url)
 
     def check_for_errors(self, response):
         try:
             response.raise_for_status()
         except requests.exceptions.HTTPError as exc:
-            raise AmigoCloudError(exc.message, exc.response)
+            raise AmigoCloudError(str(exc), exc.response)
 
-    def authenticate(self, token):
+    def authenticate(self, token, project_url=None):
```

Comment:
What happens if project_url is set but the token is not a project token ? Is AmigoCloudError thrown ?